synapse/federation/sender/__init__.py

```diff
@@ -137,6 +144,84 @@ async def get_replication_rows(
         raise NotImplementedError()
 
 
+@attr.s
+class _PresenceQueue:
+    """A queue of destinations that need to be woken up due to new presence
+    updates.
+
+    Staggers waking up of per destination queues to ensure that we don't attempt
+    to start TLS connections with many hosts all at once, leading to pinned CPU.
+    """
+
+    # The maximum duration in seconds between queuing up a destination and it
+    # being woken up.
+    _MAX_TIME_IN_QUEUE = 30.0
+
+    # The maximum duration in seconds between waking up consecutive destination
+    # queues.
+    _MAX_DELAY = 0.1
+
+    sender: "FederationSender" = attr.ib()
+    clock: Clock = attr.ib()
+    queue: "OrderedDict[str, Literal[None]]" = attr.ib(factory=OrderedDict)
```

Comment:
```suggestion
    queue: OrderedDict[str, Literal[None]] = attr.ib(factory=OrderedDict)
```

Since none of the types in here have their imports conditioned on `TYPE_CHECKING`

Reply:
FWIW I thought this was needed as `OrderdDict` wasn't generic in 3.6, but that doesn't look to be true.

Reply:
(Just saying that as that is the other reason to use quoted types)